webpack/JobWizard/steps/HostsAndInputs/index.js

```diff
@@ -1,66 +1,159 @@
+    }));
+  };
+
+  const clearSearch = () => {
+    dispatch(resetData(hostsController, hostQuerySearchID));
+    setHostsSearchQuery('');
+  };
   return (
-    <>
+    <div className="target-hosts-and-inputs">
       <WizardTitle title={WIZARD_TITLES.hostsAndInputs} />
       <Form>
         <FormGroup fieldId="host_selection">
-          <SelectField
-            fieldId="host_methods"
-            options={hostMethods}
-            setValue={setHostMethod}
-            value={hostMethod}
-          />
-          <SelectedChips
-            selected={selectedHosts}
-            setSelected={setSelectedHosts}
-          />
+          <InputGroup>
+            <SelectField
+              className="target-method-select"
+              toggleIcon={<FilterIcon />}
+              fieldId="host_methods"
+              options={Object.values(hostMethods)}
+              setValue={setHostMethod}
+              value={hostMethod}
+            />
+            {hostMethod === hostMethods.searchQuery && (
+              <HostSearch
+                setValue={setHostsSearchQuery}
+                value={hostsSearchQuery}
+              />
+            )}
+            {hostMethod === hostMethods.hosts && (
+              <HostSelect
+                selectedHosts={selectedHosts}
+                setSelectedHosts={setSelectedHosts}
+                apiKey={HOSTS}
+                url="/api/hosts?per_page=100"
+                placeholderText={__('Filter by hosts')}
+              />
+            )}
+            {withKatello && hostMethod === hostMethods.hostCollections && (
+              <HostSelect
+                selectedHosts={selectedHostCollections}
+                setSelectedHosts={setSelectedHostCollections}
+                apiKey={HOST_COLLECTIONS}
+                url="/katello/api/host_collections?per_page=100"
+                placeholderText={__('Filter by host collections')}
+              />
+            )}
+            {hostMethod === hostMethods.hostGroups && (
+              <HostSelect
+                selectedHosts={selectedHostGroups}
+                setSelectedHosts={setSelectedHostGroups}
+                apiKey={HOST_GROUPS}
+                url="/api/hostgroups?per_page=100"
+                placeholderText={__('Filter by host groups')}
+              />
+            )}
+          </InputGroup>
         </FormGroup>
-        <span>
+        <SelectedChips
+          selectedHosts={selectedHosts}
+          setSelectedHosts={setSelectedHosts}
+          selectedHostCollections={selectedHostCollections}
+          setSelectedHostCollections={setSelectedHostCollections}
+          selectedHostGroups={selectedHostGroups}
+          setSelectedHostGroups={setSelectedHostGroups}
+          hostsSearchQuery={hostsSearchQuery}
+          clearSearch={clearSearch}
+        />
+        <Text>
           {__('Apply to')}{' '}
           <Button variant="link" isInline>
             {selectedHosts.length} {__('hosts')}
           </Button>
```

Comment:
This looks like a link (coloring, cursor change) but does nothing on click.

Reply:
https://github.com/theforeman/foreman_remote_execution/pull/611